v2/tools/generator/internal/codegen/pipeline/create_resource_extension_types.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/functions"
+)
+
+// CreateResourceExtensionsStageID is the unique identifier of this stage
+const CreateResourceExtensionsStageID = "createResourceExtensions"
+
+func CreateResourceExtensions(localPath string, idFactory astmodel.IdentifierFactory) Stage {
+	return MakeLegacyStage(
+		CreateResourceExtensionsStageID,
+		"create Resource Extensions for each resource type",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+
+			extendedResourceTypes := make(astmodel.Types)
+			extendedResourceTypesMapping := make(map[astmodel.TypeName][]astmodel.TypeName)
```

Comment:
minor: A comment describing what this map is between would be helpful.

I think (from reading the code) it's a map of the new extension type name, to all of the resource type names (across versions) which the extension applies to.